CustomerOrderDataBase.java

```diff
@@ -462,15 +473,61 @@ private void deleteOrder() {
             }
         }
 
+        // If no orders were found/returned based on the user's entered email,
+        // then notify user and redirect to Main Menu Selection Screen
+        if (itemSearchingFor == null) {
+            System.out.println("        ! NOTHING FOUND/EMPTY !");
+            System.out.println("\n* Please enter 'x' to be redirected to the Main Menu Selection Screen");
+            System.out.println("--------------------------------------------");
+            String exitAttribute = sc.next();
+            if(exitAttribute.equalsIgnoreCase("x")) {
+                displayMenue();
+            }
+            else{
+                while(!exitAttribute.equalsIgnoreCase("x")) {
+                    System.out.println("Sorry please enter 'x' to be redirected! ");
+                    exitAttribute = sc.next();
+                    if(exitAttribute.equalsIgnoreCase("x")) {
+                        displayMenue();
```

Comment:
This would appear to run correctly but it's actually calling displayMenu inside of deleteOrder, so once the menu gets a QUIT, it is going to return to the first deleteOrder method on the stack. I think you could fix this by having a return instead of displayMenu.